transport.go

```diff
@@ -93,13 +92,15 @@ type Transport interface {
 	// Note: from a data transfer symantic standpoint, it doesn't matter if the
 	// request is push or pull -- OpenChannel is called by the party that is
 	// intending to receive data
-	OpenChannel(ctx context.Context,
+	OpenChannel(
+		ctx context.Context,
 		dataSender peer.ID,
 		channelID ChannelID,
 		root ipld.Link,
 		stor ipld.Node,
-		doNotSendCids []cid.Cid,
-		msg Message) error
+		channel ChannelState,
```

Comment:
sidebar +++1 for this change generally. I've never liked the specificity of doNodeSendCids. Technically, we can probably drop a bunch of other parameters if we do this (I believe root/selector/sender/channelID can all be extracted from channel state)

Reply:
I see that wouldn't quite work for the code here just yet (you're counting on channel == nil being evidence of a not-restart), but a great thing for future refactor!